#**Proyecto 1 - Parte2**
##**Elaborado por:** Antonella Arcos


In [4]:
# Importar una librería
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [35]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/CODING_DOJO/PTCienciaDeDatos/PandasParaManipulaciónDeDatos/sales_predictions.csv')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [ ]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

En la variable peso del producto hay 1463 datos faltantes y en el tamaño de la tienda en cuanto a la superficie total que cubre hay 2410 datos faltantes

In [47]:
df

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
0              FDA15        9.300  ...  Supermarket Type1          3735.1380
1              DRC01        5.920  ...  Supermarket Type2           443.4228
2              FDN15       17.500  ...  Supermarket Type1          2097.2700
3              FDX07       19.200  ...      Grocery Store           732.3800
4              NCD19        8.930  ...  Supermarket Type1           994.7052
...              ...          ...  ...                ...                ...
8518           FDF22        6.865  ...  Supermarket Type1          2778.3834
8519           FDS36        8.380  ...  Supermarket Type1           549.2850
8520           NCJ29       10.600  ...  Supermarket Type1          1193.1136
8521           FDN46        7.210  ...  Supermarket Type2          1845.5976
8522           DRG01       14.800  ...  Supermarket Type1           765.6700

[8523 rows x 12 columns]

##A continuación hago algunas tablas para entender si los missing values son debido a algun atributo en especifico

In [17]:
df.groupby(['Outlet_Establishment_Year'])[['Item_Weight','Outlet_Size','Item_Identifier']].count()
#Se compara con Item Identifier porque es una variable que tiene observaciones completas
#Se ve que en el año 1985 ningun item tiene el peso
#Se ve que en los años 1998, 2002 y 2007 no hay el tamaño de los locales
#La data perdida no responde a un valor especifico de los datos, puede ser porque en esos años donde hay novedades no se recolecto información o se perdio con el tiempo


Item_Weight  Outlet_Size  Item_Identifier
Outlet_Establishment_Year                                           
1985                                 0         1463             1463
1987                               932          932              932
1997                               930          930              930
1998                               555            0              555
1999                               930          930              930
2002                               929            0              929
2004                               930          930              930
2007                               926            0              926
2009                               928          928              928

In [19]:
df.groupby(['Outlet_Type'])[['Item_Weight','Outlet_Size','Item_Identifier']].count()
#Se compara con Item Identifier porque es una variable que tiene observaciones completas
#En grocery store falta más de la mitad de las observaciones del peso del producto y el tamaño de la tienda
#En el Supermarket Type1 faltan observaciones del tamaño de la tienda
#En el Supermarket Type3 no hay datos sobre el peso del producto

Item_Weight  Outlet_Size  Item_Identifier
Outlet_Type                                                 
Grocery Store              555          528             1083
Supermarket Type1         5577         3722             5577
Supermarket Type2          928          928              928
Supermarket Type3            0          935              935

In [21]:
#Voy a hacer una imputacion de datos faltantes con una interpolación lineal


In [48]:
df['Outlet_Size']=df.fillna(df.mode().iloc[0])
#Se lleno los vacios con la moda

In [45]:
df['Item_Weight']=df.loc[0:8523, 'Item_Weight'].interpolate(method = 'linear')
#Se hizo una interpolacion lineal

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                8523 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                8523 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB
